lib/components/SchematicViewer.tsx

```diff
@@ -44,10 +49,30 @@ export const SchematicViewer = ({
   debug = false,
   clickToInteractEnabled = false,
   colorOverrides,
+  spiceSimulationEnabled = false,
 }: Props) => {
   if (debug) {
     enableDebug()
   }
+  const [showSpiceOverlay, setShowSpiceOverlay] = useState(false)
+
+  const spiceString = useMemo(() => {
+    if (!spiceSimulationEnabled) return null
+    try {
+      return getSpiceFromCircuitJson(circuitJson)
+    } catch (e) {
+      console.error("Failed to generate SPICE string", e)
+      return null
+    }
+  }, [JSON.stringify(circuitJson), spiceSimulationEnabled])
```

Comment:
you've got to compute a circuitJsonKey, it's annoying and error prone but stringifying for a key on every render is a lot
```
const circuitJsonKey = `${circuitJson?.length || 0}_${(circuitJson as any)?.editCount || 0}`
```

```suggestion
  }, [circuitJsonKey, spiceSimulationEnabled])
```